kola/tests/devcontainer/devcontainer.go

```diff
@@ -0,0 +1,282 @@
+
+echo "Fetching developer container from ${IMAGE_URL}"
+# Stolen from copy_from_buildcache in ci_automation_common.sh. Not
+# using --output-dir option as this seems to be quite a new addition
+# and curl on older version of Flatcar does not understand it.
+curl --fail --silent --show-error --location --retry-delay 1 --retry 60 \
+        --retry-connrefused --retry-max-time 60 --connect-timeout 20 \
+        --remote-name "${IMAGE_URL}"
+
+# The image file takes over 6Gb after normal unpacking, but a lot of
+# it is just zeros. Use cp --sparse=always to avoid unnecessary disk
+# space waste. Especially that we may not have 6Gb of disk space
+# available.
+cp --sparse=always <(lbzcat flatcar_developer_container.bin.bz2) flatcar_developer_container.bin
+
+# PORTAGE_BINHOST and EXPECTED_VERSION are meant to be propagated to
+# the dev container as environment variables.
+PORTAGE_BINHOST=$(process_template '{{ .BinhostURLTemplate }}' "${ARCH}" "${VERSION}")
+EXPECTED_VERSION="${FLATCAR_RELEASE_VERSION}"
+
+# These directories (USR_SRC_DIR and VAR_TMP_DIR) are meant to be used
+# by the dev container to store files generated during emerging kernel
+# sources and making the modules_prepare job.
+#
+# Previously tmpfs was used, but under qemu we might not have enough
+# memory.
+workdir="${PWD}/dev-container-workdir-${RANDOM}"
+USR_SRC_DIR="${workdir}/src"
+VAR_TMP_DIR="${workdir}/tmp"
+mkdir -p "${USR_SRC_DIR}" "${VAR_TMP_DIR}"
+`)
+
+	systemdNspawnScriptBody = trimLeftSpace(`
+sudo systemd-nspawn \
+        --console=pipe \
+        --setenv=PORTAGE_BINHOST="${PORTAGE_BINHOST}" \
+        --setenv=EXPECTED_VERSION="${EXPECTED_VERSION}" \
+        --bind-ro=/lib/modules \
+        --bind-ro=/home/core/dev-container-script \
+        --image=flatcar_developer_container.bin \
+        --machine=flatcar-developer-container \
+        --bind="${USR_SRC_DIR}:/usr/src" \
+        --bind="${VAR_TMP_DIR}:/var/tmp" \
+        /bin/bash /home/core/dev-container-script
+`)
+
+	dockerScriptBody = trimLeftSpace(`
+# TODO: It would much much better if we provided dev-container as a
+# docker image on ghcr.io.
+
+offset=$(parted flatcar_developer_container.bin unit b print 2>/dev/null | grep 'Start' --after-context=1 | tail --lines=1 | awk '{ print $2 }' | head --bytes=-2)
+mkdir root
+sudo mount -o loop,ro,offset="${offset}" flatcar_developer_container.bin root
+sudo tar -C root -czf dev-container-image.tar.gz .
+sudo umount root
+rm -f flatcar_developer_container.bin
+docker import dev-container-image.tar.gz dev-container:42
+sudo rm -f dev-container-image.tar.gz
+stat /home/core/dev-container-script >&2
+ls -la /home/core >&2
+
+docker run \
+        --env PORTAGE_BINHOST="${PORTAGE_BINHOST}" \
+        --env EXPECTED_VERSION="${EXPECTED_VERSION}" \
+        --mount type=bind,source=/lib/modules,target=/lib/modules,readonly=true \
+        --mount type=bind,source=/home/core/dev-container-script,target=/home/core/dev-container-script,readonly=true \
```

Comment:
```suggestion
        -v /home/core/dev-container-script:/home/core/dev-container-script:z \
```
See https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label on how to grant access by relabeling

Reply:
It basically modifies the label on the host, so I went with calling restorecon. Let's see if it does the trick.

Reply:
Also, I'm following this: https://github.com/moby/moby/issues/32579#issuecomment-788389016